### DISCUSSION TITLE
How to specify required System Dependencies when deploying via Streamlit Cloud

### DISCUSSION THREAD

**Discussion by nhtkid** [Unstructured-IO/unstructured#1069] [Q&A]
I love unstructured.io. It is so powerful and so easy to use.
I literally only need two lines of code to load different type of data.
```
from langchain.document_loaders import UnstructuredFileLoader
loader = UnstructuredFileLoader("state_of_the_union.txt")
loader.load()
```

But when I deploy my code via Streamlit Cloud, how do I specify these following system dependencies?

> Install the following system dependencies if they are not already available on your system. Depending on what document types you're parsing, you may not need all of these.
libmagic-dev (filetype detection)
poppler-utils (images and PDFs)
tesseract-ocr(images and PDFs)
libreoffice (MS Office docs)
pandoc (EPUBs)

Thanks!

**Comment by nhtkid** [comment]
I have created a packages.txt based on the streamlit recommendation below with all the required packages.
[https://docs.streamlit.io/streamlit-community-cloud/deploy-your-app/app-dependencies](url)

`
ibmagic-dev
poppler-utils
tesseract-ocr
libreoffice
pandoc
`
It looks like the deployment is successfully. However, I could only upload pdf/docx/txt.
If I upload pptx or png, I keeps getting (IndexError: list index out of range).

I am not sure whether the loader is not able to load, or after loading, it was not able to convert to embeddings.
FYI I am using OpenAI embeddings and ChromaDB.

**Reply by cragwolfe** [reply]
> If I upload pptx or png, I keeps getting (IndexError: list index out of range).

My best guess is file detection went awry. Also, typo in your first line, missing the `l` in`libmagic-dev`.

To test theory, I'd try `unstructured/partition/pptx.py::partition_pptx` and `unstructured/partition/image.py::partition_image` directly.
